src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1438,11 +1438,24 @@ private void getFieldInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_MET
                     }
                     else
                     {
-                        helperId = ReadyToRunHelperId.GetNonGCStaticBase;
+                        var owningType = field.OwningType;
+                        if ((field.Context.IsWellKnownType(owningType, WellKnownType.IntPtr) ||
```

Comment:
Looking at `owningType.Category` might be more efficient, but this is unlikely to be a problem either way.